<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>650</width>
    <height>513</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="TabWidgetOperationMode" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <property name="documentMode">
       <bool>false</bool>
      </property>
      <property name="tabsClosable">
       <bool>false</bool>
      </property>
      <property name="tabBarAutoHide">
       <bool>false</bool>
      </property>
      <widget class="QWidget" name="tabMain">
       <attribute name="title">
        <string>主面板</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="0" column="0">
         <widget class="FrontMainPanel" name="widgetMain" native="true"/>
        </item>
        <item row="1" column="0">
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="FrontSlot" name="widgetSlot1" native="true"/>
          </item>
          <item row="0" column="1">
           <widget class="FrontSlot" name="widgetSlot2" native="true"/>
          </item>
          <item row="0" column="2">
           <widget class="FrontSlot" name="widgetSlot3" native="true"/>
          </item>
          <item row="0" column="3">
           <widget class="FrontSlot" name="widgetSlot4" native="true"/>
          </item>
          <item row="0" column="4">
           <widget class="FrontSlot" name="widgetSlot5" native="true"/>
          </item>
          <item row="0" column="5">
           <widget class="FrontSlot" name="widgetSlot6" native="true"/>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="FrontSafetyPanel" name="tabSafety">
       <attribute name="title">
        <string>安全I/O</string>
       </attribute>
      </widget>
      <widget class="FrontMessageShow" name="tabHistory">
       <attribute name="title">
        <string>歷史訊息</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3"/>
      </widget>
      <widget class="FrontAxisParameter" name="tabAxis">
       <attribute name="title">
        <string>伺服參數</string>
       </attribute>
      </widget>
      <widget class="FrontCylinderPanel" name="tabCylinder">
       <attribute name="title">
        <string>汽缸參數</string>
       </attribute>
      </widget>
      <widget class="FrontUnitPanel" name="tabUnit">
       <attribute name="title">
        <string>單站控制</string>
       </attribute>
      </widget>
      <widget class="FrontIoOverride" name="tabIO">
       <attribute name="title">
        <string>I/O監視</string>
       </attribute>
      </widget>
      <widget class="FrontSignal" name="tabSignal">
       <attribute name="title">
        <string>信號參數</string>
       </attribute>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>650</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuAuth">
    <property name="title">
     <string>權限</string>
    </property>
    <addaction name="actionOperator"/>
    <addaction name="actionEngineer"/>
    <addaction name="actionDeveloper"/>
   </widget>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>傳輸</string>
    </property>
    <widget class="QMenu" name="menuUpload">
     <property name="title">
      <string>Upload</string>
     </property>
     <addaction name="actionReadCylinder"/>
     <addaction name="actionReadAxis"/>
     <addaction name="actionReadCommand_Block"/>
     <addaction name="actionReadSignal"/>
     <addaction name="actionReadUnit"/>
     <addaction name="actionReadAll"/>
    </widget>
    <widget class="QMenu" name="menuDownload">
     <property name="title">
      <string>Download</string>
     </property>
     <addaction name="actionWriteCylinder"/>
     <addaction name="actionWriteAxis"/>
     <addaction name="actionWriteCommand_Block"/>
     <addaction name="actionWriteSignal"/>
     <addaction name="actionWriteUnit"/>
     <addaction name="actionWriteAll"/>
    </widget>
    <addaction name="menuUpload"/>
    <addaction name="separator"/>
    <addaction name="menuDownload"/>
   </widget>
   <addaction name="menuAuth"/>
   <addaction name="menu"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionExit">
   <property name="text">
    <string>結束</string>
   </property>
  </action>
  <action name="actionOperator">
   <property name="text">
    <string>Operator</string>
   </property>
  </action>
  <action name="actionUpload_all">
   <property name="text">
    <string>Upload all</string>
   </property>
  </action>
  <action name="actionDownload_all">
   <property name="text">
    <string>Download all</string>
   </property>
  </action>
  <action name="actionReadCylinder">
   <property name="text">
    <string>Cylinder</string>
   </property>
  </action>
  <action name="actionReadAxis">
   <property name="text">
    <string>Axis</string>
   </property>
  </action>
  <action name="actionReadCommand_Block">
   <property name="text">
    <string>Command block</string>
   </property>
  </action>
  <action name="actionReadSignal">
   <property name="text">
    <string>Signal</string>
   </property>
  </action>
  <action name="actionReadUnit">
   <property name="text">
    <string>Unit configuration</string>
   </property>
  </action>
  <action name="actionReadAll">
   <property name="text">
    <string>All</string>
   </property>
  </action>
  <action name="actionWriteCylinder">
   <property name="text">
    <string>Cylinder</string>
   </property>
  </action>
  <action name="actionWriteAxis">
   <property name="text">
    <string>Axis</string>
   </property>
  </action>
  <action name="actionWriteCommand_Block">
   <property name="text">
    <string>Command block</string>
   </property>
  </action>
  <action name="actionWriteSignal">
   <property name="text">
    <string>Signal</string>
   </property>
  </action>
  <action name="actionWriteUnit">
   <property name="text">
    <string>Unit configuration</string>
   </property>
  </action>
  <action name="actionWriteAll">
   <property name="text">
    <string>All</string>
   </property>
  </action>
  <action name="actionEngineer">
   <property name="text">
    <string>Engineer</string>
   </property>
  </action>
  <action name="actionDeveloper">
   <property name="text">
    <string>Developer</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>FrontAxisParameter</class>
   <extends>QWidget</extends>
   <header>frontaxisparameter.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>FrontCylinderPanel</class>
   <extends>QWidget</extends>
   <header>frontcylinderpanel.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>FrontUnitPanel</class>
   <extends>QWidget</extends>
   <header>frontunitpanel.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>FrontMainPanel</class>
   <extends>QWidget</extends>
   <header>frontmainpanel.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>FrontIoOverride</class>
   <extends>QWidget</extends>
   <header>frontiooverride.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>TabWidgetOperationMode</class>
   <extends>QTabWidget</extends>
   <header>tabwidgetoperationmode.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>FrontMessageShow</class>
   <extends>QWidget</extends>
   <header>frontmessageshow.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>FrontSafetyPanel</class>
   <extends>QWidget</extends>
   <header>frontsafetypanel.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>FrontSlot</class>
   <extends>QWidget</extends>
   <header>frontslot.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>FrontSignal</class>
   <extends>QWidget</extends>
   <header>frontsignal.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
